model.py

```diff
@@ -1516,9 +1516,14 @@ def notify(self, emailer, url_for):
             to_address = to_address[7:]
         deadline = None
 
-        validation, is_new = get_one_or_create(
-            _db, Validation, resource=resource
-        )
+        # Make sure there's a Validation object associated with this
+        # Resource.
+        if resource.validation is None:
+            resource.validation, is_new = create(_db, Validation)
+        else:
+            is_new = False
+        validation = resource.validation
+
```

Comment:
This is the bug fix. The `validations` table doesn't have a `resource_id`. Instead the one-to-one relationship goes the other way: `resources.validation_id`. This means that this `get_one_or_create` call isn't reliable -- "find the Validation for this Resource" is complicated in a way that "find the Resource for this Validation" wouldn't be.

I never figured out why this particular refactoring caused the bug to show up, but I'm happy with this change.